ci_framework/roles/hci_prepare/tasks/phase2.yml

```diff
@@ -0,0 +1,104 @@
+---
+# Copyright Red Hat, Inc.
+# All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+- name: Ensure directories
+  ansible.builtin.file:
+    path: "{{ item }}"
+    state: directory
+  loop:
+    - "{{ cifmw_hci_prepare_basedir }}/artifacts"
+    - "{{ cifmw_hci_prepare_dataplane_dir }}"
+
+- name: Create Ceph secrets and configure Nova
+  environment:
+    PATH: "{{ cifmw_path }}"
+    KUBECONFIG: "{{ cifmw_openshift_kubeconfig }}"
+  when: not cifmw_hci_prepare_dryrun
+  block:
+    - name: Create ceph config secret
+      ci_script:
+        output_dir: "{{ cifmw_hci_prepare_basedir }}/artifacts"
+        script: "oc apply -f {{ cifmw_hci_prepare_ceph_secret_path }}"
+
+    - name: Set Ceph FSID fact
+      ansible.builtin.set_fact:
+        cifmw_hci_prepare_ceph_fsid: "{{ (lookup('template', cifmw_hci_prepare_ceph_secret_path)|from_yaml).data['ceph.conf'] \
```

Comment:
If this is only needed for the next task you can skip creating this fact and use it directly as a var in the next task.

Reply:
Yes, correct. I can update that in a follow up change if no major changes would be required for this PR. is that ok?

Reply:
In the end, it is going to be used to create cinder backend configuration too. So it would be at least 2 times.